sechub-cli/src/mercedes-benz.com/sechub/cli/job.go

```diff
@@ -53,24 +53,44 @@ func waitForSecHubJobDone(context *Context) (status jobStatusResult) {
 			break
 		}
 
-		// PROGRESS bar ... 50 chars with dot, then next line...
-		if newLine {
-			sechubUtil.PrintIfNotSilent(strings.Repeat(" ", 29), context.config.quiet)
-			newLine = false
+		if !context.config.quiet {
+			// Progress dots
+			cursor = printProgressDot(cursor)
 		}
-		sechubUtil.PrintIfNotSilent(".", context.config.quiet)
-		if cursor++; cursor == 50 {
-			sechubUtil.PrintIfNotSilent("\n", context.config.quiet)
-			cursor = 0
-			newLine = true
-		}
-		time.Sleep(time.Duration(context.config.waitNanoseconds))
+
+		time.Sleep(time.Duration(waitNanoseconds))
+		waitNanoseconds = computeNextWaitInterval(waitNanoseconds, context.config.waitNanoseconds)
 	}
 	sechubUtil.PrintIfNotSilent("\n", context.config.quiet)
 
 	return status
 }
 
+// Print progress dots ... 50 dots, then next line...
+func printProgressDot(cursor int) int {
+	if cursor == 0 {
+		// initial identation
+		fmt.Print(strings.Repeat(" ", 29))
+	}
+
+	fmt.Print(".")
+
+	if cursor++; cursor == 50 {
```

Comment:
This statement is strange. What happens here? Is the `cursor` incremented by one and afterwards the equality check `cursor == 50` happens?

```suggestion
        cursor++
	if cursor == 50 {
```

Reply:
It's valid Go syntax. :)
I've checked the documentation: "A statement can precede conditionals; any variables declared in this statement are available in all branches."
Assignments inside the `if` are most widely used to declare variables that exist only in the scope of the `if..else` structure.
This is here not the case, so moved `cursor++` as proposed.